File: back-end/src/main/java/br/com/srv/controllers/LoginController.java
package br.com.srv.controllers;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import br.com.srv.models.requests.LoginRequest;
import br.com.srv.models.requests.TokenRequest;
import br.com.srv.models.responses.LoginResponse;
import br.com.srv.services.LoginService;
import jakarta.validation.Valid;

@RestController
@RequestMapping("/login")
public class LoginController {
	
    private static final Logger logger = LoggerFactory.getLogger(LoginController.class);
    
    @Autowired
    private LoginService loginService;
	
	@PostMapping("/validar-login")
	public ResponseEntity<LoginResponse> validarLogin(@RequestBody @Valid LoginRequest login){
		logger.info("Executando a LoginService/validarLogin");		
		LoginResponse response = loginService.validarLogin(login);
		return ResponseEntity.status(response.getStatus()).body(response);
	}
	
	@PostMapping("/validar-token")
	public ResponseEntity<Boolean> validarToken(@RequestBody @Valid TokenRequest tokenRequest){		
		logger.info("Executando a LoginService/validarToken");		
		return ResponseEntity.ok(loginService.validarToken(tokenRequest));
		
	}
}

File: bff/src/controllers/LoginController.ts
import { Request, Response } from "express";
import { LoginRequest } from "../models/LoginRequest";
import { decriptografia, logger } from "../utils/Utils";
import * as LoginService from '../services/LoginService';
import { TokenRequest } from "../models/TokenRequest";

const log = logger;

export const validarLogin = async (req: Request, res: Response) => {
    try {
        const login: LoginRequest = JSON.parse(decriptografia(req.body.hash));
        log.info("Executando a LoginService.validarLogin")     
        const retorno: any = await LoginService.validarLogin(login);
        if (retorno.status === 200) {
            res.status(retorno.status).json(retorno);
        } else {
            res.status(retorno.status).json(retorno.mensagem);
        }
    } catch (error) {
        log.error(`Erro: ${error}`);
        res.status(500).json(`Erro: ${error}`);
    }
}

export const validarToken = async (req:Request, res: Response) =>{
    try{
        const body:TokenRequest = req.body;
        log.info("Executando a LoginService.validarToken")
        const retorno: any = await LoginService.validarToken(body);        
        res.status(200).json(retorno);
    }catch(error){
        log.error(`Erro ao validar o token`);
        res.status(500).json(`Erro: ${error}`);
    }
}